src/main/java/com/atlassian/db/replica/api/AuroraCluster.java

```diff
@@ -0,0 +1,33 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.internal.aurora.AuroraReplicasDiscoverer;
+import com.atlassian.db.replica.spi.DatabaseCluster;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.stream.Stream;
+
+public final class AuroraCluster implements DatabaseCluster {
+    private final AuroraReplicasDiscoverer discoverer;
+
+    public AuroraCluster(SqlCall<Connection> connection) throws SQLException {
+        discoverer = new AuroraReplicasDiscoverer((connection.call()));
+    }
+
+    @Override
+    public Stream<Database> getReplicas() throws SQLException {
+
+        return discoverer.fetchReplicasServerIds().stream().map(it -> new Database() {
+            @Override
+            public String getUuid() {
+                return it;
+            }
+
+            @Override
+            public Connection getConnection() {
```

Comment:
I think it should return a supplier. It depends on the client, but this way we could cache connections to known replicas and fetch only the newly discovered replicas connections.

Reply:
With supplier  the customer can define  Eager/lazy initialization